astropy/stats/circstats.py

```diff
@@ -0,0 +1,481 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module contains simple functions for dealing with circular statistics, for
+instance, mean, variance, standard deviation, correlation coefficient, and so
+on. This module also cover tests of uniformity, e.g the Rayleigh and V0 tests.
+The Maximum Likelihood Estimator for the Von Mises distribution along with the
+Cramer-Rao Lower Bounds are also implemented. Almost all of the implementations
+are based on reference [1], which is also the basis for the R package
+'CircStats'.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+__all__ = ['circmean','circvar', 'circmoment', 'circcorrcoef', 'rayleightest',
+           'vtest', 'vonmisesmle', 'vonmisescrlb']
+__doctest_requires__ = {'vtest': ['scipy.stats'],
+                        'vonmisescrlb': ['scipy.special']}
+
+
+def _components(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized rectangular components
+    # of the circular data.
+    data = np.asarray(data)
+
+    assert type(p) is int
```

Comment:
Is there a reason why `p` and only `p` is checked for the correct type? Since this is a hidden function, is this check really necessary?